### DISCUSSION TITLE
Unable to find Azure Carbon Optimization API

### DISCUSSION THREAD

**Discussion by vwake7** [stackql/stackql#405] [Q&A]
@jeffreyaven I am not able to find the Service for Carbon in any of the Azure Providers. Am I missing something.  https://management.azure.com/providers/Microsoft.Carbon/carbonEmissionReports?api-version=2023-04-01-preview

**Comment by jeffreyaven** [comment]
> @jeffreyaven I am not able to find the Service for Carbon in any of the Azure Providers. Am I missing something.  https://management.azure.com/providers/Microsoft.Carbon/carbonEmissionReports?api-version=2023-04-01-preview

Hey @vwake7 it's possible that that service is not part of AutoRest which is the base user to derive the Azure specs, let me investigate

**Comment by jeffreyaven** [comment]
Hi @vwake7 i cant seem to find the spec for the Carbon service, we generate the Azure provider using [stackql-azure-openapi](https://github.com/stackql/stackql-azure-openapi) which gets its source data from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs/tree/main/specification), MSFT sometimes doesn't publish specs for some services, if you can find a spec for this (in any format) we could generate this service.  Alternatively if you can run a few of the operations you need through the `az` cli with the debug option we could reverse engineer the service (at least a few resources and methods), let us know.

**Reply by vwake7** [reply]
Hey @jeffreyaven If you are expecting a spec similar to Compute service https://github.com/Azure/azure-rest-api-specs/tree/main/specification/compute, I will try to create one.

**Reply by jeffreyaven** [reply]
hey @vwake7 if you can generate an openapi spec we can use [openapisaurus](https://github.com/stackql/openapisaurus) to prepare the stackql extensions
